examples/H2_Analysis/h2_setup_optimize.py

```diff
@@ -287,91 +78,79 @@ def setup_cost_calcs(scenario,hybrid_plant,electrolyzer_size_mw,wind_size_mw,sol
-        hybrid_plant.solar.financial_model.FinancialParameters.debt_percent = scenario['Debt Equity']
-        if scenario['ITC']:
-            hybrid_plant.solar.financial_model.TaxCreditIncentives.itc_fed_percent = 26
+        hybrid_plant.pv._financial_model.FinancialParameters.analysis_period = scenario['Useful Life']
+        hybrid_plant.pv._financial_model.FinancialParameters.debt_percent = scenario['Debt Equity']
+        if scenario['ITC Available']:
+            hybrid_plant.pv._financial_model.TaxCreditIncentives.itc_fed_percent = 26
         else:
-            hybrid_plant.solar.financial_model.TaxCreditIncentives.itc_fed_percent = 0
+            hybrid_plant.pv._financial_model.TaxCreditIncentives.itc_fed_percent = 0
 
     if wind_size_mw > 0:
-        hybrid_plant.wind.financial_model.FinancialParameters.analysis_period = scenario['Useful Life']
-        hybrid_plant.wind.financial_model.FinancialParameters.debt_percent = scenario['Debt Equity']
-        if scenario['PTC']:
+        hybrid_plant.wind._financial_model.FinancialParameters.analysis_period = scenario['Useful Life']
+        hybrid_plant.wind._financial_model.FinancialParameters.debt_percent = scenario['Debt Equity']
+        if scenario['PTC Available']:
             ptc_val = 0.022
         else:
             ptc_val = 0.0
 
         interim_list = list(
-            hybrid_plant.wind.financial_model.TaxCreditIncentives.ptc_fed_amount)
+            hybrid_plant.wind._financial_model.TaxCreditIncentives.ptc_fed_amount)
         interim_list[0] = ptc_val
-        hybrid_plant.wind.financial_model.TaxCreditIncentives.ptc_fed_amount = tuple(interim_list)
-        hybrid_plant.wind.system_model.Turbine.wind_turbine_hub_ht = scenario['Tower Height']
+        hybrid_plant.wind._financial_model.TaxCreditIncentives.ptc_fed_amount = tuple(interim_list)
+        hybrid_plant.wind._system_model.Turbine.wind_turbine_hub_ht = scenario['Tower Height']
 
     hybrid_plant.ppa_price = 0.05
     hybrid_plant.wind.system_capacity_kw = wind_size_mw*1000
     return hybrid_plant
 
 
 def calculate_h_lcoe_continuous(bat_model,electrolyzer_size,wind_capacity_mw,solar_capacity_mw,battery_storage_mwh,battery_charge_rate,battery_discharge_rate,
-                        scenario,buy_from_grid=False,sell_to_grid=False,solar_cost_multiplier=1.0):
+                        scenario,buy_from_grid=False,sell_to_grid=False,solar_cost_multiplier=1.0,interconnection_size_mw=False):
 
+    if interconnection_size_mw:
+        interconnection_size = interconnection_size_mw
+    else:
+        interconnection_size = electrolyzer_size * 1000.0
     wind_size_mw = scenario['Turbine Rating']
-    hybrid_plant = setup_power_calcs(scenario,wind_size_mw,solar_capacity_mw,battery_storage_mwh,battery_discharge_rate) 
+    hybrid_plant = setup_power_calcs(scenario,wind_size_mw,solar_capacity_mw,battery_storage_mwh,battery_discharge_rate,interconnection_size) 
```

Comment:
The function parameters for setup_power_calcs is:

```
setup_power_calcs(scenario,wind_size_mw,solar_size_mw,storage_size_mwh,storage_size_mw,interconnection_size_mw)
```

but here you have battery_discharge_rate instead of storage_size_mw

Reply:
Yes this is fine for now. Leaving open the option of having different charge and discharge rates in the future.